services/storage/store.go

```diff
@@ -261,6 +302,16 @@ func (s *Store) TagKeys(ctx context.Context, req *datatypes.TagKeysRequest) (cur
 		if found := reads.HasFieldValueKey(expr); found {
 			return nil, errors.New("field values unsupported")
 		}
+		if found := reads.ExprHasKey(expr, fieldKey); found {
```

Comment:
This should have a new InfluxQL test for `SHOW TAG KEYS` that hits this case.

Reply:
Confirmed that InfluxQL is unaffected by this change, adding a `panic` doesn't cause any existing tests to fail